From f6df524130b6acc707a08a7406276e5509a10d76 Mon Sep 17 00:00:00 2001
From: Yongxin Liu <yongxin.liu@windriver.com>
Date: Thu, 13 Jun 2024 10:26:50 +0000
Subject: [PATCH 16/24] rdk/snr: add default_groups and ATTRIBUTE_GROUPS for
 kernel >= v5.18

kernel commit cdb4f26a63c3 ("kobject: kobj_type: remove default_attrs").
kernel commit f2f37f58b1b9 ("sysfs.h: add ATTRIBUTE_GROUPS() macro").

Upstream-Status: Inappropriate [RDK is released in tarball only]

Signed-off-by: Yongxin Liu <yongxin.liu@windriver.com>
---
 .../crypto/qat/qat_common/adf_uio_control.c      | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/klm/qat/qat/drivers/crypto/qat/qat_common/adf_uio_control.c b/klm/qat/qat/drivers/crypto/qat/qat_common/adf_uio_control.c
index 088623e..589a36f 100644
--- a/klm/qat/qat/drivers/crypto/qat/qat_common/adf_uio_control.c
+++ b/klm/qat/qat/drivers/crypto/qat/qat_common/adf_uio_control.c
@@ -84,6 +84,10 @@ static struct attribute *accel_default_attrs[] = {
 	NULL,
 };
 
+#if KERNEL_VERSION(5, 18, 0) <= LINUX_VERSION_CODE
+ATTRIBUTE_GROUPS(accel_default);
+#endif
+
 static ssize_t accel_show(struct kobject *kobj, struct attribute *attr,
 			  char *buf);
 static const struct sysfs_ops accel_sysfs_ops = {
@@ -178,6 +182,10 @@ static struct attribute *bundle_default_attrs[] = {
 	NULL,
 };
 
+#if KERNEL_VERSION(5, 18, 0) <= LINUX_VERSION_CODE
+ATTRIBUTE_GROUPS(bundle_default);
+#endif
+
 static ssize_t bundle_show(struct kobject *kobj, struct attribute *attr,
 			   char *buf);
 static const struct sysfs_ops bundle_sysfs_ops = {
@@ -227,7 +235,11 @@ static void adf_uio_sysfs_cleanup(struct adf_accel_dev *accel_dev)
 static struct kobj_type accel_ktype = {
 	.release = accel_kobject_free,
 	.sysfs_ops = &accel_sysfs_ops,
+#if KERNEL_VERSION(5, 18, 0) <= LINUX_VERSION_CODE
+	.default_groups = accel_default_groups,
+#else
 	.default_attrs = accel_default_attrs,
+#endif
 };
 
 int adf_uio_sysfs_create(struct adf_accel_dev *accel_dev)
@@ -280,7 +292,11 @@ int adf_uio_sysfs_create(struct adf_accel_dev *accel_dev)
 static struct kobj_type bundle_ktype = {
 	.release = bundle_kobject_free,
 	.sysfs_ops = &bundle_sysfs_ops,
+#if KERNEL_VERSION(5, 18, 0) <= LINUX_VERSION_CODE
+	.default_groups = bundle_default_groups,
+#else
 	.default_attrs = bundle_default_attrs,
+#endif
 };
 
 int adf_uio_sysfs_bundle_create(struct pci_dev *pdev,
-- 
2.39.3

